src/convert.js

```diff
@@ -48,7 +52,7 @@ function convertFeature(features, geojson, options) {
             for (i = 0; i < coords.length; i++) {
                 geometry = [];
                 convertLine(coords[i], geometry, tolerance, false);
-                features.push(createFeature(geojson.id, 'LineString', geometry, geojson.properties));
+                features.push(createFeature(id, 'LineString', geometry, geojson.properties));
```

Comment:
It looks like when `lineMetrics` is turned on, when multiline features are split into multiple features, they won't get the same id with either of id options active, while we need a single id retained across the parts.

Reply:
The `for`-loop uses the same `id` for all LineString features split here. For both options, the `id` is computed L32-L37 and would carry for all the line strings.